core/network/src/config.rs

```diff
@@ -388,8 +388,18 @@ pub enum Secret<K> {
 	///   * `secp256k1::SecretKey`: An unencoded 32 bytes Secp256k1 secret key.
 	///   * `ed25519::SecretKey`: An unencoded 32 bytes Ed25519 secret key.
 	File(PathBuf),
-	/// Always generate a new secret key `K`.
-	New
+}
+
+impl Default for Secret<ed25519::SecretKey> {
```

Comment:
Maybe `Default` is not the best trait here. I'd expect to always get the same value when calling `Default` which isn't the case here. Maybe just add a `random` or `generate` method to `Secret` to make it more explicit?

Reply:
Added a new method for each variant